llvm/lib/Transforms/Instrumentation/InstrProfiling.cpp

```diff
@@ -170,6 +170,29 @@ cl::opt<bool> SkipRetExitBlock(
     "skip-ret-exit-block", cl::init(true),
     cl::desc("Suppress counter promotion if exit blocks contain ret."));
 
+static cl::opt<bool> SampledInstr("sampled-instr", cl::ZeroOrMore,
+                                  cl::init(false),
+                                  cl::desc("Do PGO instrumentation sampling"));
+
+static cl::opt<unsigned> SampledInstrPeriod(
+    "sampled-instr-period",
+    cl::desc("Set the profile instrumentation sample period. For each sample "
+             "period, the 'sampled-instr-burst-duration' number of consecutive "
```

Comment:
A fixed number of consecutive samples will be record. The number is controlled by 'sampled-instr-burst-duration' flag.

Reply:
ack.